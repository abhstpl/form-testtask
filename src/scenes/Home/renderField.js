import React from 'react';

const renderField = ({ input, label, type, placeholder,validationError, meta: { touched, error, warning } }) => {

  return(
  <div>
    <label>{placeholder} </label>
    <div>
      <input {...input} placeholder={placeholder} type={type}  className={"form-control"}/>
    </div>
  </div>
)
};

export default renderField;