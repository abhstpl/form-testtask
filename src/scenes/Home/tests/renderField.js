import React from 'react';
import { shallow } from 'enzyme';
import renderField from '../renderField';

describe('Test on renderField', () => {
	it('should render fields', () => {
		const meta = {
				touched: true,  
				error: 'error',
				warning: 'warning'
		};
		const wrapper = shallow(<renderField input="text" placeholder="test" type="text" validationError={true} className={"form-control"} meta={meta}/>);
		expect(wrapper).toMatchSnapshot();
		});

  it('should render fields', () => {
    const meta = {
          touched: false
    };
    const wrapper = shallow(<renderField input="test" placeholder="test" type="text" validationError={false} className={"form-control"} meta={meta}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render fields', () => {
    const meta = {
			touched: false,
			error: 'error',
			warning: 'warning'
    };
		const wrapper = shallow(<renderField input="test" placeholder="test" type="text" validationError={false} meta={meta}/>);
		expect(wrapper).toMatchSnapshot();
    });
});