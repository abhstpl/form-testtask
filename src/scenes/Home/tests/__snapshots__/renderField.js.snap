// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on renderField should render fields 1`] = `
<renderField
  className="form-control"
  input="text"
  meta={
    Object {
      "error": "error",
      "touched": true,
      "warning": "warning",
    }
  }
  placeholder="test"
  type="text"
  validationError={true}
/>
`;

exports[`Test on renderField should render fields 2`] = `
<renderField
  className="form-control"
  input="test"
  meta={
    Object {
      "touched": false,
    }
  }
  placeholder="test"
  type="text"
  validationError={false}
/>
`;

exports[`Test on renderField should render fields 3`] = `
<renderField
  input="test"
  meta={
    Object {
      "error": "error",
      "touched": false,
      "warning": "warning",
    }
  }
  placeholder="test"
  type="text"
  validationError={false}
/>
`;
